### PR TITLE
bootstrap binary patchelf

### PR DESCRIPTION
**Pull request by haampie**
- Bootstrap binary patchelf
- Do not blindly use system patchelf
- Require recent patchelf

### CODE REVIEW

**Comment by haampie** [comment]
So, this works. Can we move the binaries to the official mirror?

File: lib/spack/spack/bootstrap.py [hunk 1]
```diff
@@ -791,14 +791,40 @@ def patchelf_root_spec():
     # TODO: patchelf is restricted to v0.13 since earlier versions have
     # TODO: bugs that we don't to deal with, while v0.14 requires a C++17
     # TODO: which may not be available on all platforms.
-    return _root_spec("patchelf@0.13.1:0.13.99")
+    return _root_spec("patchelf@0.13.1:")
```
Comment (alalazo) [inline lib/spack/spack/bootstrap.py:792]: The TODO is misleading with this PR. Can you remove that?

File: lib/spack/spack/bootstrap.py [hunk 2]
```diff
@@ -791,14 +791,40 @@ def patchelf_root_spec():
...
+
+
+def verify_patchelf(patchelf):
+    """Older patchelf versions can produce broken binaries in many, many
+    different ways, we should guard against that."""
```
Comment (alalazo) [inline lib/spack/spack/bootstrap.py]: (minor) Can you add an Args: entry, specifying that this expects an `Executable` object?

File: share/spack/bootstrap/github-actions-v0.2/patchelf.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "verified": [
```
Comment (alalazo) [inline share/spack/bootstrap/github-actions-v0.2/patchelf.json]: Can you setup a:
```
share/spack/bootstrap/github-actions-v0.3
```
folder and move `patchelf.json in there + symlink `clingo.json` and `gnupg.json`?

File: share/spack/bootstrap/github-actions-v0.2/metadata.yaml [hunk 4]
```diff
@@ -3,6 +3,6 @@ description: |
   Buildcache generated from a public workflow using Github Actions.
   The sha256 checksum of binaries is checked before installation.
 info:
-  url: https://mirror.spack.io/bootstrap/github-actions/v0.2
+  url: http://harmenstoppels.nl/spack-bootstrap/
```
Comment (alalazo) [inline share/spack/bootstrap/github-actions-v0.2/metadata.yaml]: The URL should now be `https://mirror.spack.io/bootstrap/github-actions/v0.3`

**Review by alalazo — CHANGES_REQUESTED** [review]
The most important request is to use a v0.3 and not override our current mirror. There are a few other minor comments. The buildcache with patchelf is at:
```
https://mirror.spack.io/bootstrap/github-actions/v0.3
```

File: share/spack/bootstrap/github-actions-v0.2/patchelf.json (hunk 3 above)
Comment (haampie) [reply share/spack/bootstrap/github-actions-v0.2/patchelf.json]: I don't see the point of having `v0.1`, `v0.2`, `v0.3`, ... Can't I just rename the directory to 0.3?

Comment (haampie) [reply share/spack/bootstrap/github-actions-v0.2/patchelf.json]: Ah, it's referenced in the config... let's leave it for compat then. But at some point we could prune those guys...

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks!

**Pull request merged**